go/vt/vttablet/tabletserver/vstreamer/uvstreamer.go

```diff
@@ -368,19 +369,44 @@ func (uvs *uvstreamer) Stream() error {
 		uvs.sendTestEvent("Copy Done")
 	}
 	log.V(2).Infof("Starting replicate in uvstreamer.Stream()")
-	vs := newVStreamer(uvs.ctx, uvs.cp, uvs.se, uvs.sh, mysql.EncodePosition(uvs.pos), mysql.EncodePosition(uvs.stopPos), uvs.filter, uvs.vschema, uvs.send)
-	uvs.vs = vs
+	vs := newVStreamer(uvs.ctx, uvs.cp, uvs.se, uvs.sh, mysql.EncodePosition(uvs.pos), mysql.EncodePosition(uvs.stopPos), uvs.filter, uvs.getVSchema(), uvs.send)
+
+	uvs.setVs(vs)
 	return vs.Stream()
 }
 
+func (uvs *uvstreamer) lock(msg string) {
+	log.V(2).Infof("Acquiring uvs lock: %s", msg)
```

Comment:
Once debugged, we should get rid of these logs. I believe they hinder the compiler optimization.